### PR TITLE
feat: migrate to OTEL and enable metrics by default

### PR DESCRIPTION
**Pull request by mutianf**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)
- [ ] Rollback plan is reviewed and LGTMed
- [ ] All new data plane features have a completed end to end testing plan

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

**Review by rkaregar — APPROVED** [review]

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]

File: google-cloud-bigtable-deps-bom/pom.xml [hunk 1]
```diff
@@ -77,6 +77,18 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-bom</artifactId>
+        <version>1.34.1</version>
```
[the remark this answers is not in the document]
Comment (mutianf) [reply google-cloud-bigtable-deps-bom/pom.xml:83]: I can't bump the version until we pick up the latest version bump in the shared libraries

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+          .collect(Collectors.toSet());
+
+  public static BigtableCloudMonitoringExporter create(
+      String projectId, @Nullable Credentials credentials) throws IOException {
+    MetricServiceSettings.Builder settingsBuilder = MetricServiceSettings.newBuilder();
+    CredentialsProvider credentialsProvider;
+    if (credentials == null) {
+      credentialsProvider = NoCredentialsProvider.create();
+    } else {
+      credentialsProvider = FixedCredentialsProvider.create(credentials);
```
[the remark this answers is not in the document]
Comment (mutianf) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java:125]: Hmm I don't think this works?

[the remark this answers is not in the document]
Comment (igorbernstein2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java:124]: ?

**Review by igorbernstein2 — APPROVED** [review]

**Pull request merged**